### PR TITLE
UPE reads customer details from checkout fields

### LINKED ISSUE #1981: UPE Missing Customer details on transactions page
As noted in this [Github thread](https://github.com/Automattic/woocommerce-payments/issues/1828#issuecomment-850055235), the customer details are not displayed when checking out with UPE. The most likely issue is that the customer details are not being properly added to the UPE transaction. 

 **To Reproduce**
- Enable UPE checkout with the `_wcpay_feature_upe` set to 1 in the `wp_options` table
- Checkout with UPE payment method
- Go to the order in admin and click on the payment transaction link at the top (pi_xxxx)
- View the transaction details and note the missing customer data
- Click on Payments > Transactions and from the list of transactions the UPE checkout will be missing customer data

Markup 2021-06-01 at 10 26 08
Markup 2021-06-01 at 10 26 36

### PR DESCRIPTION
**Pull request by FangedParakeet**
Fixes #1981 

#### Changes proposed in this Pull Request

# UPE Reads Customer Details from Checkout Fields

> As noted in this Github thread, the customer details are not displayed when checking out with UPE. The most likely issue is that the customer details are not being properly added to the UPE transaction.
> ...
> We have gotten the latest feature description from Stripe for how to implement the additional billing details and hide the appropriate fields from UPE to avoid the customer having to enter information twice. We can now hide the fields that are duplicated in the billing details from UPE and instead provide the details from the billing details when confirming the payment. The fields in particular that should be hidden are: name, country, and postal code.
>
> In addition to the fields to hide, we should also provide as many of the billing fields as possible when confirming the intent. In particular: name, email, phone, and address fields.

This PR allows checkout fields to be read by the UPE upon the point of payment confirmation, so that we can add the customer's details to the payment transaction details. This should only affect the UPE when it is loaded on the checkout page: in other pages, such as the Order Pay page or the Add Payment Method page, there are no customer details present, so these pages will continue to work as they have been.

Once concern that I noticed is that even though we tell Stripe to hide certain fields when we mount the UPE, I noticed that with Sofort the country field remained present inside the UPE. If I remove the country argument from the `payment_method_data` billing details when I confirm the payment, I receive this error.

> You specified "never" for fields.billing_details.address.country when creating the payment Element, but did not pass confirmParams.payment_method_data.billing_details.address.country when calling stripe.confirmPayment or stripe.confirmSetup. If you opt out of collecting data via the payment Element using the fields option, the data must be passed at confirm-time.

This implies that I'm setting the flag for Stripe to hide this field correctly, but for whatever reason, it doesn't seem to hide the field, so I'm not entirely clear what's going on here. 

Everything else seems to be working smoothly enough.

#### Testing instructions

1. Ensure you have pulled the changes in this PR and the latest version of the Payments Server. Also, please enable the UPE gateway using the WCPay Dev Plugin or simply set _wcpay_feature_upe to a value of 1 in your wp_options table. 
2. Add a product to your cart and proceed to the checkout screen. 
3. Complete the checkout process as usual.
4. Either go to **Payments > Transactions** and select the completed transaction or view the completed order and click on the payment intent ID near the top of the page to be transported to the details page for this latest transaction.
Payment intent ID on order page
_Payment intent ID on order page_

5. Confirm that customer details from checkout are present on the transaction.
6. Confirm that you can still add a payment method to a customer account or pay for an order through the Order Pay page without any issues.
7. Make a wish for lucky step number 7.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/checkout/classic/upe.js [hunk 1]
```diff
@@ -192,6 +205,32 @@ jQuery( function ( $ ) {
...
+	 *
+	 * @param {Object} fields Object mapping checkout billing fields to values.
+	 * @return {Object} Stripe formatted `billing_details` object.
+	 */
+	const getBillingDetails = ( fields ) => {
+		return {
+			name: (
+				fields.billing_first_name +
+				' ' +
+				fields.billing_last_name
```
Comment (frosso) [inline client/checkout/classic/upe.js:219]: (minor) we could use string a template here
```suggestion
				`${fields.billing_first_name} ${fields.billing_last_name}`
```

**Review by frosso — APPROVED** [review]
I saw both names on the ticket so I thought about checking this out.

Do I have to say my wish out loud or shall I keep it for me?

File: client/checkout/classic/upe.js (hunk 1 above)
Comment (FangedParakeet) [reply client/checkout/classic/upe.js:219]: Nice, I like it. Done,

Comment (frosso) [reply client/checkout/classic/upe.js]: Thanks for considering it!
TBH, I haven't ensured that linter would be happy with it 👀

**Comment by FangedParakeet** [comment]
@frosso keep your wish to yourself, but let us know if it comes true. Maybe you can open a new issue for your mystery wish and we can all track progress over there. 

Thanks for the review! If you fancy, I'll let you click "Update branch" before I merge and then all three of us can proudly have had a part in this ticket.

**Comment by frosso** [comment]
@FangedParakeet lol thanks, now that I clicked "Update branch" I really feel like we're all holding hands in a 3-person circle of joy 🧑‍🤝‍🧑🧑‍🤝‍🧑
Go get merged in peace code, become part of The One and teach the angels how to fly! 👼

**Pull request merged**